tests/integration/webclient/src/test/java/io/helidon/tests/integration/webclient/RequestTest.java

```diff
@@ -126,6 +126,49 @@ public void testFollowRedirectInfinite() {
         }
     }
 
+    @Test
+    public void testTemporaryRedirect() throws Exception {
+        webClient.put()
+                .path("/redirect/temporary")
+                .submit(JSON_NEW_GREETING)
+                .thenAccept(response -> assertThat(response.status().code(), is(204)))
```

Comment:
Do not do any assertions in the reactive flow. Simply block until you get response and validate that.

Reply:
I will not write this to all the cases here, but change it everywhere you use it.

Reply:
Currently in those tests there is a flow of events happening, should I split this all into blocking calls?

Reply:
Exactly what I meant :-)

Reply:
Ok)